### PR TITLE
Feature/validation

### PR DESCRIPTION
**Pull request by mgupta0421**
This PR implements the following functionalities -
1- Validation of query in terms of Syntax.
2- Progress bar to show the search query is running. Fix for Issue# 58.
3- Metric results are displayed in the table. Fix for Issue# 56.
4- Styling changes and fix for Issue# 55, 57 and 60.

### CODE REVIEW

File: frontend/src/components/landingPage/MenuStepTwo.vue [hunk 1]
```diff
@@ -11,7 +11,7 @@
 					<option>=</option>
 				</select>
 				<input class="form-control" id="valueInput" type="text" placeholder="Input Value" v-model="selectedValue" @input="sendValue">  
-				<h6 id="nanInformation" v-show="notANumber">Value should be numerical</h6>           
+				<h6 id="nanInformation" v-show="notANumber">Number Please</h6>
```
Comment (bhermann) [inline frontend/src/components/landingPage/MenuStepTwo.vue]: I liked the former text better... much more polite. :)

File: frontend/src/components/landingPage/SearchPart.vue [hunk 2]
```diff
@@ -55,38 +68,100 @@ export default {
...
-      finalQueryToReset: false,
+      savedQuery: "", //query from queryMenu will be saved here
+      readyToSearchQuery: "", //finalQuery from the query component will be saved here
       headers: [
-        { text: "ArtifactId", align: "left", value: "metadata.artifactId" },
-        { text: "GroupId", value: "metadata.groupId" },
-        { text: "Source", value: "metadata.source" },
-        { text: "Version", value: "metadata.version" }
+        { text: "ArtifactId", align: "left", value: "metadata.artifactId", width: "10%" },
+        { text: "GroupId", value: "metadata.groupId", width: "10%" },
```
Comment (bhermann) [inline frontend/src/components/landingPage/SearchPart.vue]: Could you reorder groupId and artifactId so groupId comes first? That's a bit more natural for people knowing Maven.

**Comment by bhermann** [comment]
Also will try running the application tomorrow...

**Comment by mgupta0421** [comment]
Taken care of scala codacy issue.

File: frontend/src/components/landingPage/MenuStepTwo.vue (hunk 1 above)
Comment (SViktorS) [reply frontend/src/components/landingPage/MenuStepTwo.vue]: I changed it back. Also decreased the font size a little bit.

File: frontend/src/components/landingPage/SearchPart.vue (hunk 2 above)
Comment (SViktorS) [reply frontend/src/components/landingPage/SearchPart.vue]: Done

**Comment by bhermann** [comment]
I had to rerun npm install for this PR.
Interestingly, I had to install Python 2 to do this... weird... Which new dependency triggered that?
Is there an alternative?

**Comment by bhermann** [comment]
Compiler warning not addressed:
[warn] /.../HomeController.scala:71:7: match may not be exhaustive.
[warn] It would fail on the following input: Failure((x: Throwable forSome x not in org.parboiled2.ParseError))
[warn]       parsingResult match {
[warn]       ^

Technically this means you could have runtime errors.

**Comment by bhermann** [comment]
Confirming positive results for #58, #55, #57, and #60. 
Solution for #56 is the bare minimum. Only one metric can be displayed. Please address this in a future version, but we already talked about this. 
Validation of the syntax currently helps only professional users. Please work on a solution (for a future version) for users less proficient in parser technology. :)

**Pull request merged**